File: POO JAVA/Aula10/src/aula10/Professor.java
package aula10;

public class Professor extends Pessoa{
    
    //Atributos da SubClasse
    private String especialidade;
    private float salario;

    //Metodos Acessores e Modificadores
    public String getEspecialidade() {
        return especialidade;
    }
    public void setEspecialidade(String especialidade) {
        this.especialidade = especialidade;
    }
    public float getSalario() {
        return salario;
    }
    public void setSalario(float salario) {
        this.salario = salario;
    }
    
    //Metodo da SubClasse
    public void receberAumento(float aument){
        this.salario += aument;
    }
}

File: POO PHP/Aula10/Professor.php
<?php
require_once 'Pessoa.php';

class Professor extends Pessoa{

    //Atributos da Classe
    private $especialidade;
    private $salario;
    
    //Metodos Acessores e Modificadores
    public function getEspecialidade() {
        return $this->especialidade;
    }
    public function getSalario() {
        return $this->salario;
    }
    public function setEspecialidade($especialidade): void {
        $this->especialidade = $especialidade;
    }
    public function setSalario($salario): void {
        $this->salario = $salario;
    }

    //Metodo da Classe
    public function receberAumento($aumento) {
        $this->salario += $aumento;
    }
}
